src/api/general-services/pipeline-manage/constructors/create-new-step.js

```diff
@@ -99,11 +98,7 @@ const createNewStep = (context, step, args) => {
                     task,
                   ],
                   env: [
-                    {
-                      name: 'SAMPLE_ID',
-                      ...perSample && { 'value.$': '$.sampleUuid' },
-                      ...!perSample && { value: 'all' },
-                    },
+                    ...perSample ? [{ name: 'SAMPLE_ID', 'value.$': '$.sampleUuid' }] : [],
```

Comment:
The default `SAMPLE_ID` if none is specified (i.e. the task is not sample-specific) should be no env variable at all (pipeline will convert that into an empty string, which is what R expects).